### PR TITLE
Offline scan without chroot in RPM*_probe, textfile*_probe

### PR DESCRIPTION
**Pull request by WesleyCeraso**

### CODE REVIEW

File: src/OVAL/probes/independent/textfilecontent.c [hunk 1]
```diff
@@ -382,13 +388,24 @@ int probe_main(probe_ctx *ctx, void *arg)
 	pfd.pattern = pattern;
 	pfd.filename_ent = filename_ent;
 	pfd.ctx = ctx;
+	
+	if (OSCAP_GSYM(offline_mode) & PROBE_OFFLINE_OWN)
+	{
+		strcpy(path_with_root, getenv("OSCAP_PROBE_ROOT"));
```
Comment (mpreisler) [inline src/OVAL/probes/independent/textfilecontent.c]: Use strncpy, you know the max size of the target buffer...

File: src/OVAL/probes/independent/textfilecontent.c [hunk 2]
```diff
@@ -382,13 +388,24 @@ int probe_main(probe_ctx *ctx, void *arg)
 	pfd.pattern = pattern;
 	pfd.filename_ent = filename_ent;
 	pfd.ctx = ctx;
+	
+	if (OSCAP_GSYM(offline_mode) & PROBE_OFFLINE_OWN)
+	{
```
Comment (mpreisler) [inline src/OVAL/probes/independent/textfilecontent.c]: code style

File: src/OVAL/probes/independent/textfilecontent.c [hunk 3]
```diff
@@ -327,16 +327,22 @@ static int process_file(const char *path, const char *filename, void *arg)
...
-	probe_setoption(PROBEOPT_OFFLINE_MODE_SUPPORTED, PROBE_OFFLINE_CHROOT);
 	return NULL;
 }
 
 int probe_main(probe_ctx *ctx, void *arg)
 {
 	SEXP_t *path_ent, *filename_ent, *line_ent, *behaviors_ent, *filepath_ent, *probe_in;
 	char *pattern;
+	char path_with_root[PATH_MAX];
+	int root_len = 1;
```
Comment (mpreisler) [inline src/OVAL/probes/independent/textfilecontent.c]: root_len can be unsigned

File: src/OVAL/probes/independent/textfilecontent.c [hunk 4]
```diff
@@ -382,13 +388,24 @@ int probe_main(probe_ctx *ctx, void *arg)
 	pfd.pattern = pattern;
 	pfd.filename_ent = filename_ent;
 	pfd.ctx = ctx;
+	
+	if (OSCAP_GSYM(offline_mode) & PROBE_OFFLINE_OWN)
+	{
+		strcpy(path_with_root, getenv("OSCAP_PROBE_ROOT"));
+		// minimum length should be 1
+		if (!root_len)
```
Comment (mpreisler) [inline src/OVAL/probes/independent/textfilecontent.c]: this condition can never be met, why is it even here?

File: src/OVAL/probes/independent/textfilecontent.c [hunk 5]
```diff
@@ -382,13 +388,24 @@ int probe_main(probe_ctx *ctx, void *arg)
...
+			root_len = 1;
+	}
 
 	if ((ofts = oval_fts_open(path_ent, filename_ent, filepath_ent, behaviors_ent, probe_ctx_getresult(ctx))) != NULL) {
 		while ((ofts_ent = oval_fts_read(ofts)) != NULL) {
 			if (ofts_ent->fts_info == FTS_F
 			    || ofts_ent->fts_info == FTS_SL) {
+				// -1 overlaps the /
+				strcpy(path_with_root + root_len - 1, ofts_ent->path);
```
Comment (mpreisler) [inline src/OVAL/probes/independent/textfilecontent.c]: strncpy

**Comment by WesleyCeraso** [comment]
@mpreisler thanks for the review. It's updated now

File: src/OVAL/probes/independent/textfilecontent54.c [hunk 6]
```diff
@@ -498,12 +504,23 @@ int probe_main(probe_ctx *ctx, void *arg)
...
+
+		if (path_with_root[root_len - 1] == FILE_SEPARATOR)
+			--root_len;
+	}
+
 	if ((ofts = oval_fts_open(path_ent, file_ent, filepath_ent, bh_ent, probe_ctx_getresult(ctx))) != NULL) {
 		while ((ofts_ent = oval_fts_read(ofts)) != NULL) {
 			if (ofts_ent->fts_info == FTS_F
 			    || ofts_ent->fts_info == FTS_SL) {
+				strcpy(path_with_root + root_len, ofts_ent->path);
```
Comment (jan-cerny) [inline src/OVAL/probes/independent/textfilecontent54.c]: There is a strcpy forgotten.

**Comment by WesleyCeraso** [comment]
@jan-cerny thanks. Fixed

**Comment by jan-cerny** [comment]
Great! Looks OK.

@matusmarhefka @dahaic Do you think that we should have a test for this new feature?

**Comment by dahaic** [comment]
@jan-cerny yeah, it would be great to have some tests for this feature :)

**Pull request merged**